src/main/java/org/opensearch/flowframework/rest/RestGetWorkflowStepAction.java

```diff
@@ -69,13 +71,12 @@ protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient
                 );
             }
 
-            ActionRequest request = new ActionRequest() {
-                @Override
-                public ActionRequestValidationException validate() {
-                    return null;
-                }
-            };
-            return channel -> client.execute(GetWorkflowStepAction.INSTANCE, request, ActionListener.wrap(response -> {
+            Map<String, String> params = request.hasParam(WORKFLOW_STEP)
+                ? Map.of(WORKFLOW_STEP, request.param(WORKFLOW_STEP))
+                : Collections.emptyMap();
+
+            WorkflowRequest workflowRequest = new WorkflowRequest(null, null, params);
```

Comment:
why not create a WorkflowStepRequest class? Not a must since we already have a few different APIs using this, just seeming like its becoming a catch all for many APIs

Reply:
I didn't have to change any code in the `WorkflowRequest` class to use it here. Instead of adding another request class when the same thing can be achieved using the existing one. If we have additional requirements beyond what's already in the current `WorkflowRequest` class, then it definitely makes sense to introduce another request class.

Reply:
I am okay with this for now, I personally prefer having a Request class for each API (provision, get, create) instead of the multiple constructors but its already there, agree if we make a change that requires another constructor we can potentially refactor it then.